### DISCUSSION TITLE
Unless for of !if

### DISCUSSION THREAD

**Discussion by ryuheechul** [lmorg/murex#631] [Ideas]
at first glance,`!if` feels a bit strange to me and I know some languages use `unless` for negative conditions. So I would consider adding `unless' keyword for that (if not (or !if 😉) replacing).

**Comment by lmorg** [comment]
I can see why `!if` might read a little weird but there is some method to the madness.

A lot of functions within Murex support the bang sigil as a shorthand for doing the opposite. In a lot of cases it's not a logical opposite but more of a functional distinction. For example `!export` will unset environmental variables. Likewise `!alias` will undefine an alias. So `!if` falls into the same camp as them and while the `!` prefix might not be the most intuitive to read, it should be intuitive to find because it follows the same design patterns.

I'm a little reluctant to add `unless`, ...erm..., unless (sorry, no pun intended) there is demand for that. However it would be very easy for you to alias it yourself if you wanted to use that in your own code:
```
alias unless=!if
```

**Reply by ryuheechul** [reply]
Thanks for the reply (and the pun)! I actually agree with all of your view especially about the demand part, so time will tell! I will give it a go with aliasing 😄
